Sources/PackageGraph/PackageGraph.swift

```diff
@@ -54,6 +55,13 @@ enum PackageGraphError: Swift.Error {
 
 /// A collection of packages.
 public struct PackageGraph {
+    /// The variable inputs to a package graph that are relevant when trying to reload individual packages from an existing graph.
+    public struct Inputs {
+        public let binaryArtifacts: [PackageIdentity: [String: BinaryArtifact]]
+        public let shouldCreateMultipleTestProducts: Bool
+        public let createREPLProduct: Bool
+    }
```

Comment:
I suggest moving binaryArtifacts out of here since its the same "conceptual level" as rootPackages, rootDependencies and dependencies. then we make shouldCreateMultipleTestProducts, createREPLProduct a "configuration" of sorts? not too crazy about storing it tho at the graph level. should this all be driven from the workspace?

Reply:
Moving the `binaryArtifacts` sounds reasonable to me.

For the other properties, I see three options:
- Store them "somewhere" in the `PackageGraph` object as done here
- Move them up to `WorkspaceConfiguration` which means they can longer be changed per `loadPackageGraph()` call
- Make them parameters of `loadPackage()` and require the client to remember them

I think the last option wouldn't be too great, so it comes down to whether we think it is valuable to be able to change them per `loadPackageGraph()` call.

Reply:
We agreed offline to move the parameters to `WorkspaceConfiguration`.